src/Conversion/ONNXToKrnl/NN/Conv.cpp

```diff
@@ -67,143 +73,164 @@ struct ONNXConvOpLowering : public ConversionPattern {
 
     // Determine the bounds for the loops over batch & channel out.
     IndexExpr iZero = LiteralIndexExpr(0);
-    ValueRange outerLoops = createKrnl.defineLoops(3);
+    IndexExpr iOne = LiteralIndexExpr(1);
+
+    SmallVector<Value, 3> lbsStorage, ubsStorage, stepsStorage;
     SmallVector<IndexExpr, 3> outerLbs = {iZero, iZero, iZero};
     SmallVector<IndexExpr, 3> outerUbs = {N, G, COPerGroup};
+    SmallVector<IndexExpr, 3> outerSteps = {iOne, iOne, iOne};
+    IndexExpr::getValues(outerLbs, lbsStorage);
+    IndexExpr::getValues(outerUbs, ubsStorage);
+    IndexExpr::getValues(outerSteps, stepsStorage);
+    ValueRange parLbs(lbsStorage);
+    ValueRange steps(stepsStorage);
+    ValueRange parUbs(ubsStorage);
     // Iterate over the outer loops
     // for n = 0 .. N:
     //   for g = 0 .. G:
     //     for coPerGroup = 0 .. COPerGroup:
     //       co = g * COPerGroup + coPerGroup;
 
-    MemRefBuilder createMemRef(createKrnl);
+    // MemRefBuilder createMemRef(create);
```

Comment:
can be removed, as well as most of the commented out code below... unless you need to dematerialize later. But from the look of it, its not needed as you are using a lambda function and instantiate very nicely below for either parallel or sequential.